pkg/endpoint/bpf.go

```diff
@@ -424,26 +424,8 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string, regenContext *Regene
 		return e.nextPolicyRevision, compilationExecuted, nil
 	}
 
-	// Anything below this point must be reverted upon failure as we are
-	// changing live BPF maps
-	createdPolicyMap := false
-
-	defer func() {
-		if reterr != nil {
-			e.Logger().WithError(err).Error("destroying BPF maps due to" +
-				" errors during regeneration")
-			if createdPolicyMap {
-				e.UnconditionalLock()
-				e.Logger().Debug("removing endpoint PolicyMap")
-				os.RemoveAll(e.PolicyMapPathLocked())
-				e.PolicyMap = nil
-				e.Unlock()
-			}
-		}
-	}()
-
 	if e.PolicyMap == nil {
-		e.PolicyMap, createdPolicyMap, err = policymap.OpenMap(e.PolicyMapPathLocked())
+		e.PolicyMap, _, err = policymap.OpenMap(e.PolicyMapPathLocked())
```

Comment:
This was the last call to `policymap.OpenMap` that made use of the returned "new map" boolean.
That can be removed from `policymap.OpenMap`.

Reply:
More generally, it seems like that the "new map" boolean returned by `bpf.OpenOrCreateMap` is not needed anymore.

Reply:
Agreed. I kept the fix minimal as we backport this to all branches.